security/backends/elasticsearch/throttling.py

```diff
@@ -0,0 +1,77 @@
+from datetime import timedelta
+
+from django.conf import settings
+from django.utils import timezone
+from django.utils.translation import ugettext as _
+from django.urls import resolve
+
+from elasticsearch_dsl import Q
+
+from ipware.ip import get_client_ip
+
+from security.enums import InputRequestSlug
+from security.throttling.validators import ThrottlingValidator
+from security.throttling.exception import ThrottlingException
+
+from .models import InputRequestLog
+
+
+class PerRequestThrottlingValidator(ThrottlingValidator):
+
+    def __init__(self, timeframe, throttle_at, description=_('Slow down')):
+        super().__init__(timeframe, throttle_at, description)
+
+    def _validate(self, request):
+        count_same_requests = InputRequestLog.search().filter(
+            Q('term', ip=get_client_ip(request)[0])
+            & Q('term', path=request.path)
+            & Q('range', start={'gte': timezone.now() - timedelta(seconds=self.timeframe)})
+            & Q('term', method=request.method.upper())
+        ).count()
+        return count_same_requests < self.throttle_at
+
+
+class LoginThrottlingValidator(ThrottlingValidator):
+
+    slug = None
+
+    def _validate(self, request):
+        count_same_requests = InputRequestLog.search().filter(
+            Q('term', ip=get_client_ip(request)[0])
+            & Q('term', path=request.path)
+            & Q('range', start={'gte': timezone.now() - timedelta(seconds=self.timeframe)})
+            & Q('slug', slug=slug)
+        ).count()
+        return count_same_requests < self.throttle_at
```

Comment:
I think that this decision logic should be implemented in parent. The child classes should only implement counting of the requests, because that is the only thing that is different across the backends.

Reply:
Yes you are right. I wil solve this in next PR. I want to do some api functions in the backends which will return number of requests for some input (some univerzal filter api). And throttling validator will be only one. This is temporary solution